interpreter/cling/lib/Interpreter/NullDerefProtectionTransformer.cpp

```diff
@@ -33,58 +33,7 @@ namespace cling {
   NullDerefProtectionTransformer::~NullDerefProtectionTransformer()
   { }
 
-  // Copied from clad - the clang/opencl autodiff project
-  class NodeContext {
-  public:
-  private:
-    typedef llvm::SmallVector<clang::Stmt*, 2> Statements;
-    Statements m_Stmts;
-  private:
-    NodeContext() {};
-  public:
-    NodeContext(clang::Stmt* s) { m_Stmts.push_back(s); }
-    NodeContext(clang::Stmt* s0, clang::Stmt* s1) {
-      m_Stmts.push_back(s0);
-      m_Stmts.push_back(s1);
-    }
-
-    bool isSingleStmt() const { return m_Stmts.size() == 1; }
-
-    clang::Stmt* getStmt() {
-      assert(isSingleStmt() && "Cannot get multiple stmts.");
-      return m_Stmts.front();
-    }
-    const clang::Stmt* getStmt() const { return getStmt(); }
-    const Statements& getStmts() const {
-      return m_Stmts;
-    }
-
-    CompoundStmt* wrapInCompoundStmt(clang::ASTContext& C) const {
-      assert(!isSingleStmt() && "Must be more than 1");
-      llvm::ArrayRef<Stmt*> stmts
-        = llvm::makeArrayRef(m_Stmts.data(), m_Stmts.size());
-      clang::SourceLocation noLoc;
-      return new (C) clang::CompoundStmt(C, stmts, noLoc, noLoc);
-    }
-
-    clang::Expr* getExpr() {
-      assert(llvm::isa<clang::Expr>(getStmt()) && "Must be an expression.");
-      return llvm::cast<clang::Expr>(getStmt());
-    }
-    const clang::Expr* getExpr() const {
-      return getExpr();
-    }
-
-    void prepend(clang::Stmt* S) {
-      m_Stmts.insert(m_Stmts.begin(), S);
-    }
-
-    void append(clang::Stmt* S) {
-      m_Stmts.push_back(S);
-    }
-  };
-
-  class IfStmtInjector : public StmtVisitor<IfStmtInjector, NodeContext> {
+  class IfStmtInjector : public StmtVisitor<IfStmtInjector, void> {
```

Comment:
While we're waiting for roottest :-) Can we rename this to something else? E.g. PointerCheckInjector?